AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/Support/Model+Sort.swift

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+
+extension Array where Element: Model {
+    mutating func sortModels(by sortBy: QuerySortDescriptor, modelSchema: ModelSchema) {
+        sort { modelSchema.comparator(model1: $0, model2: $1, sortBy: sortBy) }
+    }
+}
+
+enum ModelValueCompare<T: Comparable> {
+    case bothNil
+    case leftNil(value2: T)
+    case rightNil(value1: T)
```

Comment:
if i remember correctly, `leftNil` case means the left side "value1" is nil so the associated value is the "right side". it may clearer if
`case leftNil(rightValue: T)`